runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -132,14 +132,17 @@ inline void generateLoadJ9Class(TR::Node* node, TR::Register* j9class, TR::Regis
       switch (opValue)
          {
          case TR::checkcastAndNULLCHK:
+         case TR::monent:
```

Comment:
Why is this unconditional? It is only true if value types are enabled right?

Reply:
It looks like `opValue` couldn't be `monent` or `monexit` before (it would assert before). But now `generateLoadJ9Class` is called by `monitorEnterOrExitForValueTypeClass` which can pass in a `monent` or `monexit` node. So it can only get to these new cases if value types are enabled.

Reply:
@cathyzhyi Could we add a fatal assert to capture that fact?

Reply:
Added fatal assert as suggested.